nemo_skills/inference/model/genselect.py

```diff
@@ -60,7 +72,18 @@ def __init__(self, model: BaseModel, cfg: OnlineGenSelectConfig):
             tokenizer = None
         self.genselect_prompt = get_prompt(prompt_config=self.cfg.prompt_config, tokenizer=tokenizer)
 
-        self.semaphore = asyncio.Semaphore(self.cfg.max_concurrent_requests)
+        # Initialize the solutions if input_dir is provided
+        if self.cfg.generation_dir is not None:
+            LOG.info("Loading solutions from %s", self.cfg.generation_dir)
+            self.prompt_to_solutions_dict = self._load_solutions(self.cfg.generation_dir)
+            LOG.info("Loaded solutions for %d prompts", len(self.prompt_to_solutions_dict))
+
+        # TODO: These calculations will change for GenSelect Competition
```

Comment:
these look a bit too conservative? Need to make sure we aren't underutilizing gpus in most cases by default